chrome/content/zotero/xpcom/zotero.js

```diff
@@ -1188,6 +1219,40 @@ Components.utils.import("resource://gre/modules/osfile.jsm");
 		
 		return useProfileDir ? true : file;
 	}
+	
+	
+	this.forceNewDataDirectory = function() {
+		var win = Services.wm.getMostRecentWindow('navigator:browser');
```

Comment:
Should probably take an optional `win` parameter and use that instead of this if provided, since this is triggered from the pref window, not the main browser window.